bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsStrategy.scala

```diff
@@ -85,12 +85,30 @@ object AgsStrategy {
      * Creates a new TargetEnvironment with guide stars for each assignment in
      * the Selection.
      */
-    def applyTo(env: TargetEnvironment): TargetEnvironment =
+    def applyTo(env: TargetEnvironment): TargetEnvironment = {
+      def findMatching(gpt: GuideProbeTargets, target: SPTarget): Option[SPTarget] = {
+        def name(t: SPTarget): Option[String] =
+          Option(t.getTarget.getName).map(_.trim)
+
+        name(target).flatMap(n => gpt.getTargets.asScalaList.find(name(_).exists(_ == n)))
+      }
+
       (env /: assignments) { (curEnv, ass) =>
         val target = new SPTarget(HmsDegTarget.fromSkyObject(ass.guideStar.toOldModel))
         val oldGpt = curEnv.getPrimaryGuideProbeTargets(ass.guideProbe).asScalaOpt
-        val newGpt = oldGpt.getOrElse(GuideProbeTargets.create(ass.guideProbe)).withManualPrimary(target)
+
+        val newGpt = oldGpt.fold(GuideProbeTargets.create(ass.guideProbe, target).withExistingPrimary(target)) { gpt =>
+          // We already have guide probe targets for guide probe.
+          // Does one with the same target name already exist? If so, mark it as
+          // primary and replace it with the target we just made.  If not, add the
+          // target and mark it as primary.
+          findMatching(gpt, target).fold(gpt.withManualPrimary(target)) { existing =>
+            gpt.withExistingPrimary(existing).setPrimary(target)
```

Comment:
Yeah, it did work this way previously.

I guess that the idea is that if the user edits the AGS target and then invokes AGS, then this restores it in its new and pristine state.

We might be able to get around some syncing if we compare the underlying `ITarget`s after matching on name to see if the replacement is actually necessary (i.e. `!oldITarget.equals(newITarget)`), but this is essentially verbatim pre-BAGS behaviour.